client/v3/retry_interceptor.go

```diff
@@ -149,6 +149,17 @@ func (c *Client) streamClientInterceptor(optFuncs ...retryOption) grpc.StreamCli
 	}
 }
 
+// shouldRefreshToken checks whether there's a need to refresh the token based on the error and callOptions,
```

Comment:
lgtm @mitake 
@r-ashish thanks and one quick review comment - can you please make sure if we should use this new func here as well https://github.com/etcd-io/etcd/blob/main/client/v3/retry_interceptor.go#L250 (ref to same PR that you have mentioned)

Reply:
@spzala thanks for checking and pointing that out. I think we already have the logic to get a new token for streams call on reconnect ([retry_interceptor.go#L119](https://github.com/etcd-io/etcd/blob/main/client/v3/retry_interceptor.go#L119)) but would be good to use this new func at [retry_interceptor.go#L250](https://github.com/etcd-io/etcd/blob/main/client/v3/retry_interceptor.go#L250) as well.

Anyway, I'll update it soon.

Reply:
Updated.